process_testcase.go

```diff
@@ -270,7 +287,100 @@ func (v *Venom) runTestSteps(ctx context.Context, tc *TestCase) {
 		}
 
 		tc.computedVars.AddAll(assign)
+		}
+	}
+}
+
+//Parse and format range data to allow iterations over user data
+func parseRanged(ctx context.Context, rawStep []byte, stepVars H) (Range, error) {
+
+	//Load "range" attribute and perform actions depending on its typing
+	var ranged Range
+	json.Unmarshal([]byte(rawStep), &ranged)
+
+	switch ranged.RawContent.(type) {
+
+	//Nil means this is not a ranged data, append an empty item to force at least one iteration and exit
+	case nil:
+		ranged.Items = append(ranged.Items, RangeData{})
+		return ranged, nil
+
+	//String needs to be parsed and possibly templated
+	case string:
+		Debug(ctx, "attempting to parse \"range\" attribute")
+		rawString := ranged.RawContent.(string)
+		if len(rawString) == 0 {
+			return ranged, fmt.Errorf("\"range\" has been specified without any data")
+		}
+
+		// Try parsing already templated data
+		err := json.Unmarshal([]byte("{\"range\":"+rawString+"}"), &ranged)
+
+		// ... or fallback
+		if err != nil {
+			//Try templating and escaping data
+			Debug(ctx, "attempting to template \"range\" attribute and parse it again")
+			vars, err := DumpStringPreserveCase(stepVars)
+			if err != nil {
+				break
+			}
+			for i := range vars {
+				vars[i] = strings.ReplaceAll(vars[i], "\"", "\\\"")
+			}
+			content, err := interpolate.Do(string(rawStep), vars)
+			if err != nil {
+				break
+			}
+
+			//Try parsing data
+			err = json.Unmarshal([]byte(content), &ranged)
+			if err != nil {
+				break
+			}
+			switch ranged.RawContent.(type) {
+			case string:
+				rawString = ranged.RawContent.(string)
+				json.Unmarshal([]byte("{\"range\":"+rawString+"}"), &ranged)
+			}
+		}
 	}
+
+	//Format data
+	switch t := ranged.RawContent.(type) {
+
+	//Array-like data
+	case []interface{}:
+		Debug(ctx, "\"range\" data is array-like")
+		for index, value := range ranged.RawContent.([]interface{}) {
+			key := strconv.Itoa(index)
+			ranged.Items = append(ranged.Items, RangeData{key, value})
+		}
+
+	//Number data
+	case float64:
+		Debug(ctx, "\"range\" data is number-like")
+		upperBound := int(ranged.RawContent.(float64))
+		for i := 0; i < upperBound; i++ {
+			key := strconv.Itoa(i)
+			ranged.Items = append(ranged.Items, RangeData{key, i})
+		}
+
+	//Map-like data
+	case map[string]interface{}:
```

Comment:
```suggestion
	case map[string]interface{},map[interface{}]interface{}:
```

Reply:
Is the aim of using `map[interface{}]interface{}` for complex mapping ?

Like: 
```yaml
range:
      ? {key: foo}
      : {actual: hello, expected: hello}
      ? {key: bar}
      : {actual: world, expected: world}	
```
If that's the case I don't think it's supported from what I've seen [go.dev/play/p/4yC3E54dLGy](https://go.dev/play/p/4yC3E54dLGy) though I may have missed something

Reply:
tbh i don't remember when/how it's possible to get `map[interface{}]interface{}` out of an unmarshaler, you can skip this comment, if you prefer

Reply:
ok I'll let as it since I don't have a test case for it 
I don't think it's possible to make dynamic type assertions, so it would actually require to create a new case to handle specifically `ranged.RawContent.(map[interface{}]interface{})`